File: Nim/is_prime.nim
import math
import strutils

proc isPrime(x : int): bool =
  if x == 2:
    return true
  elif x mod 2 == 0 or x < 2:
    return false
  else:
    for i in countup(3, int(math.sqrt(float64(x)))):
      if x mod i == 0:
        return false
    return true
  

let x : int = parseInt(readline(stdin));

if isPrime(x):
  echo "Yes"
else:
  echo "No"

File: Python/is_prime.py
import math

def isPrime(x):
    if x == 2:
        return True
    elif x % 2 == 0 or x < 2:
        return False
    else:
        for i in range(3, int(math.sqrt(x)) + 1):
            if x % i == 0:
                return False
        return True
        
x = 0
x = int(input())

if isPrime(x):
    print("Yes")
else:
    print("No")
